### PR TITLE
drivers/audio: Stop thread only if kd and epd are false

### PR DESCRIPTION
**Pull request by junmin-kim**
Move process thread create time
registerprocess -> startprocess

thread terminate time
unregisterprocess -> stopprocess

### CODE REVIEW

**Comment by junmin-kim** [comment]
@Taejun-Kwon This is just a draft.
How do you think?

**Comment by seinfra** [comment]
Target : [8d2c611a8937b084670f28ee4a81fbb2aec334b5] - Code Rule Check Result:
os/drivers/audio/audio_null.c:992: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/audio/audio_null.c:1014: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [8d2c611a8937b084670f28ee4a81fbb2aec334b5] - Code Rule Check (C++) OK.

File: os/drivers/audio/audio_null.c [hunk 1]
```diff
@@ -1000,8 +1009,24 @@ static int null_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned long
 	case AUDIOIOC_STOPPROCESS: {
 		auddbg("set stop process!!\n");
 #ifdef CONFIG_AUDIO_PROCESSING_FEATURES
-		priv->speech_state = AUDIO_NULL_SPEECH_STATE_NONE;
-		priv->process_terminate = true;
+		if ((uint8_t)arg == AUDIO_SD_KEYWORD_DETECT) {
```
Comment (Taejun-Kwon) [inline os/drivers/audio/audio_null.c]: This is acceptable for null driver, but our another target board should use one thread.
So, when driver_stop or process_stop called, it should check priv->running & priv->kd, epd.
currently this is what I want.

File: os/drivers/audio/audio_null.c [hunk 2]
```diff
@@ -1000,8 +1009,24 @@ static int null_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd, unsigned long
...
+			priv->keyword_detect = false;
+		} else if((uint8_t)arg == AUDIO_SD_ENDPOINT_DETECT) {
+			priv->endpoint_detect = false;
+		}
+
+		if (!priv->keyword_detect && !priv->endpoint_detect) {
```
Comment (Taejun-Kwon) [inline os/drivers/audio/audio_null.c]: Or we can create thread at register, destroy at unregister.
Then it may need a sem_wait.

**Comment by seinfra** [comment]
Target : [4ee1c042f0512c5c56614611ee4dfdade112728b] - Code Rule Check Result:
os/drivers/audio/audio_null.c:992: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/drivers/audio/audio_null.c:1013: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Target : [4ee1c042f0512c5c56614611ee4dfdade112728b] -

**Comment by seinfra** [comment]
Target : [a608679b082ab5776913099e3765c78b15267070] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a608679b082ab5776913099e3765c78b15267070] - Code Rule Check OK.

**Pull request closed without merging**